In [33]:
import pandas as pd
import numpy as np


In [34]:
suicide = pd.read_csv('data/suicide.csv')

In [35]:
suicide.head()

country  year     sex          age  suicides_no  population  \
0  Albania  1987    male  15-24 years           21      312900   
1  Albania  1987    male  35-54 years           16      308000   
2  Albania  1987  female  15-24 years           14      289700   
3  Albania  1987    male    75+ years            1       21800   
4  Albania  1987    male  25-34 years            9      274300   

   suicides/100k pop country-year  HDI for year  gdp_for_year ($)   \
0               6.71  Albania1987           NaN      2,156,624,900   
1               5.19  Albania1987           NaN      2,156,624,900   
2               4.83  Albania1987           NaN      2,156,624,900   
3               4.59  Albania1987           NaN      2,156,624,900   
4               3.28  Albania1987           NaN      2,156,624,900   

   gdp_per_capita ($)       generation  
0                 796     Generation X  
1                 796           Silent  
2                 796     Generation X  
3                 796  G.I. Generation  
4                 796          Boomers

In [36]:
suicide.isna().sum()

country                   0
year                      0
sex                       0
age                       0
suicides_no               0
population                0
suicides/100k pop         0
country-year              0
HDI for year          19456
 gdp_for_year ($)         0
gdp_per_capita ($)        0
generation                0
dtype: int64

In [37]:
len(suicide)

27820

In [38]:
suicide["HDI for year"].value_counts()

0.888    84
0.772    84
0.713    84
0.830    72
0.793    72
         ..
0.614    12
0.728    12
0.522    12
0.879    12
0.924    12
Name: HDI for year, Length: 305, dtype: int64

In [39]:
per_country = (suicide.groupby("country").agg({"suicides_no" : sum}))

In [40]:
per_country

suicides_no
country                          
Albania                      1970
Antigua and Barbuda            11
Argentina                   82219
Armenia                      1905
Aruba                         101
...                           ...
United Arab Emirates          622
United Kingdom             136805
United States             1034013
Uruguay                     13138
Uzbekistan                  34803

[101 rows x 1 columns]

In [41]:
per_country.loc[per_country["suicides_no"].idxmax()]

suicides_no    1209742
Name: Russian Federation, dtype: int64

The country with max suicide number in a sum is Russian Federation with a total of 1209742

In [42]:
korea = pd.read_csv("data/indicators_kor.csv")

In [43]:
korea.head()

Country Name   Country ISO3        Year                    Indicator Name  \
0  #country+name  #country+code  #date+year                   #indicator+name   
1    Korea, Rep.            KOR        2000  Agricultural machinery, tractors   
2    Korea, Rep.            KOR        1999  Agricultural machinery, tractors   
3    Korea, Rep.            KOR        1998  Agricultural machinery, tractors   
4    Korea, Rep.            KOR        1997  Agricultural machinery, tractors   

    Indicator Code                 Value  
0  #indicator+code  #indicator+value+num  
1   AG.AGR.TRAC.NO                191631  
2   AG.AGR.TRAC.NO                176146  
3   AG.AGR.TRAC.NO                157888  
4   AG.AGR.TRAC.NO                131358

In [44]:
korea["Indicator Name"].value_counts()

School enrollment, primary and secondary (gross), gender parity index (GPI)    235
Total reserves (includes gold, current US$)                                    177
Mortality rate, under-5 (per 1,000 live births)                                177
Agriculture, forestry, and fishing, value added (% of GDP)                     177
Domestic credit to private sector (% of GDP)                                   176
                                                                              ... 
Female professional and technical workers (% of total)                           1
Repetition rate in Grade 2 of primary education, male (%)                        1
Repetition rate in Grade 5 of primary education, female (%)                      1
Firms offering formal training (% of firms)                                      1
Checks used to make payments (% age 15+)                                         1
Name: Indicator Name, Length: 2153, dtype: int64

In [45]:
korea_urban_dev = pd.read_csv("data/urban-development_kor.csv")

In [46]:
korea_urban_dev.head()

Country Name   Country ISO3        Year  \
0  #country+name  #country+code  #date+year   
1    Korea, Rep.            KOR        2010   
2    Korea, Rep.            KOR        2000   
3    Korea, Rep.            KOR        1990   
4    Korea, Rep.            KOR        2010   

                                      Indicator Name     Indicator Code  \
0                                    #indicator+name    #indicator+code   
1  Urban land area where elevation is below 5 met...  AG.LND.EL5M.UR.K2   
2  Urban land area where elevation is below 5 met...  AG.LND.EL5M.UR.K2   
3  Urban land area where elevation is below 5 met...  AG.LND.EL5M.UR.K2   
4  Urban land area where elevation is below 5 met...  AG.LND.EL5M.UR.ZS   

                  Value  
0  #indicator+value+num  
1           630.0079956  
2           630.0079956  
3           630.0079956  
4           0.635875186

In [47]:
korea_social_dev = pd.read_csv("data/social-development_kor.csv")

In [48]:
korea_social_dev.head()

Country Name   Country ISO3        Year  \
0  #country+name  #country+code  #date+year   
1    Korea, Rep.            KOR        2017   
2    Korea, Rep.            KOR        2016   
3    Korea, Rep.            KOR        2015   
4    Korea, Rep.            KOR        2014   

                                      Indicator Name     Indicator Code  \
0                                    #indicator+name    #indicator+code   
1  School enrollment, primary (gross), gender par...  SE.ENR.PRIM.FM.ZS   
2  School enrollment, primary (gross), gender par...  SE.ENR.PRIM.FM.ZS   
3  School enrollment, primary (gross), gender par...  SE.ENR.PRIM.FM.ZS   
4  School enrollment, primary (gross), gender par...  SE.ENR.PRIM.FM.ZS   

                  Value  
0  #indicator+value+num  
1               0.99754  
2               0.99847  
3               0.99847  
4               0.99891

In [49]:
korea_poverty = pd.read_csv("data/poverty_kor.csv")

In [50]:
korea_poverty.head()

Country Name   Country ISO3        Year                   Indicator Name  \
0  #country+name  #country+code  #date+year                  #indicator+name   
1    Korea, Rep.            KOR        2012  Income share held by second 20%   
2    Korea, Rep.            KOR        2010  Income share held by second 20%   
3    Korea, Rep.            KOR        2008  Income share held by second 20%   
4    Korea, Rep.            KOR        2006  Income share held by second 20%   

    Indicator Code                 Value  
0  #indicator+code  #indicator+value+num  
1   SI.DST.02ND.20                    13  
2   SI.DST.02ND.20                  12.9  
3   SI.DST.02ND.20                  12.8  
4   SI.DST.02ND.20                  12.9

In [51]:
korea_poverty_grouped = (korea_poverty.groupby(["Indicator Name", "Year"]).agg({"Value": sum}))

In [52]:
korea_poverty_grouped

Value
Indicator Name                                     Year                            
#indicator+name                                    #date+year  #indicator+value+num
GINI index (World Bank estimate)                   2006                        31.7
                                                   2008                        32.3
                                                   2010                          32
                                                   2012                        31.6
Income share held by fourth 20%                    2006                        23.1
                                                   2008                          23
                                                   2010                        23.1
                                                   2012                        23.2
Income share held by highest 10%                   2006                          24
                                                   2008                        24.5
                                                   2010                        24.2
                                                   2012                        23.8
Income share held by highest 20%                   2006                        39.2
                                                   2008                        39.7
                                                   2010                        39.4
                                                   2012                          39
Income share held by lowest 10%                    2006                         2.7
                                                   2008                         2.6
                                                   2010                         2.6
                                                   2012                         2.6
Income share held by lowest 20%                    2006                         7.3
                                                   2008                         7.2
                                                   2010                         7.3
                                                   2012                         7.3
Income share held by second 20%                    2006                        12.9
                                                   2008                        12.8
                                                   2010                        12.9
                                                   2012                          13
Income share held by third 20%                     2006                        17.5
                                                   2008                        17.2
                                                   2010                        17.3
                                                   2012                        17.5
Poverty gap at $1.90 a day (2011 PPP) (%)          2006                         0.2
                                                   2008                         0.2
                                                   2010                         0.2
                                                   2012                         0.1
Poverty gap at $3.20 a day (2011 PPP) (%)          2006                         0.3
                                                   2008                         0.3
                                                   2010                         0.3
                                                   2012                         0.2
Poverty gap at $5.50 a day (2011 PPP) (%)          2006                         0.5
                                                   2008                         0.5
                                                   2010                         0.7
                                                   2012                         0.5
Poverty headcount ratio at $1.90 a day (2011 PP... 2006                         0.2
                                              

In [53]:
happy_alcohol = pd.read_csv("data/HappinessAlcoholConsumption.csv")

In [54]:
happy_alcohol.head()

Country          Region Hemisphere  HappinessScore  HDI  GDP_PerCapita  \
0      Denmark  Western Europe      north           7.526  928         53.579   
1  Switzerland  Western Europe      north           7.509  943         79.866   
2      Iceland  Western Europe      north           7.501  933         60.530   
3       Norway  Western Europe      north           7.498  951         70.890   
4      Finland  Western Europe      north           7.413  918         43.433   

   Beer_PerCapita  Spirit_PerCapita  Wine_PerCapita  
0             224                81             278  
1             185               100             280  
2             233                61              78  
3             169                71             129  
4             263               133              97

In [55]:
korea = happy_alcohol.loc[happy_alcohol["Country"].str.contains("Korea")]

In [56]:
korea

Country        Region Hemisphere  HappinessScore  HDI  GDP_PerCapita  \
51  South Korea  Eastern Asia       noth           5.835  900         27.105   

    Beer_PerCapita  Spirit_PerCapita  Wine_PerCapita  
51             140                16               9

In [57]:
happy_alcohol["Country"] = happy_alcohol["Country"].replace({"South Korea": "Republic of Korea"})

In [58]:
korea = happy_alcohol.loc[happy_alcohol["Country"].str.contains("Korea")]

In [59]:
korea

Country        Region Hemisphere  HappinessScore  HDI  \
51  Republic of Korea  Eastern Asia       noth           5.835  900   

    GDP_PerCapita  Beer_PerCapita  Spirit_PerCapita  Wine_PerCapita  
51         27.105             140                16               9

In [60]:
russia = happy_alcohol.loc[happy_alcohol["Country"].str.contains("Russia")]

In [61]:
russia

Country                      Region Hemisphere  HappinessScore  \
49  Russian Federation  Central and Eastern Europe      north           5.856   

    HDI  GDP_PerCapita  Beer_PerCapita  Spirit_PerCapita  Wine_PerCapita  
49  815          8.748             247               326              73

In [62]:
noth = happy_alcohol.loc[happy_alcohol["Hemisphere"] == "noth"]

In [65]:
noth

happy_alcohol["Hemisphere"] = happy_alcohol["Hemisphere"].replace({"noth": "north"})

In [ ]:
happy_alcohol.to_csv("data/happiness_alcohol.csv")